be/src/fs/fs_starlet.cpp

```diff
@@ -0,0 +1,241 @@
+    auto format_str = format_starlet_path(uri);
+
+    if (!uri->uri.key().empty() && uri->uri.key().back() != '/') {
+        format_str.push_back('/');
+    }
+
+    staros::starlet::ObjectStorePtr object_store = g_starlet->get_store(format_str);
+    auto st = object_store->iterate_objects(format_str, cb);
+    LOG(WARNING) << "iter  dir " << format_str << " return  " << st.ToString();
+    return to_status(st);
+}
+
+Status StarletFileSystem::create_dir(const std::string& dirname) {
+    auto uri = format_starlet_path(dirname);
+    auto format_str = format_starlet_path(uri);
+    auto st = is_directory(format_str);
+    if (st.ok() && st.value()) {
+        return Status::AlreadyExist(dirname);
+    }
+    if (uri->uri.key().empty() || uri->uri.key() == "/") {
+        return Status::AlreadyExist(fmt::format("root directory already exist"));
+    }
+
+    if (uri->uri.key().back() != '/') {
+        format_str.push_back('/');
+    }
+    staros::starlet::ObjectStorePtr object_store = g_starlet->get_store(format_str);
+    auto res = object_store->create_empty_object(format_str);
+    LOG(WARNING) << "create dir " << format_str << " return  " << res.ToString();
+    return to_status(res);
+}
+
+Status StarletFileSystem::create_dir_if_missing(const std::string& dirname, bool* created) {
+    auto st = create_dir(dirname);
+    if (created != nullptr) {
+        *created = st.ok();
+    }
+    if (st.is_already_exist()) {
+        st = Status::OK();
+    }
+    return st;
+}
+
+Status StarletFileSystem::create_dir_recursive(const std::string& dirname) {
+    return create_dir_if_missing(dirname, nullptr);
+}
+
+Status StarletFileSystem::delete_dir(const std::string& dirname) {
+    auto uri = format_starlet_path(dirname);
+    auto format_str = format_starlet_path(uri);
+    staros::starlet::ObjectStorePtr object_store = g_starlet->get_store(format_str);
+    std::set<std::string> files;
+    if (uri->uri.key().empty() || uri->uri.key() == "/") {
+        return Status::NotSupported("Cannot delete root directory of StarletFS");
+    }
+    if (uri->uri.key().back() != '/') {
+        format_str.push_back('/');
+    }
+    auto cb = [&files](std::string_view file) {
+        files.emplace(file);
+        return true;
+    };
+    auto st = object_store->iterate_objects(format_str, cb);
+    LOG(WARNING) << "iter  dir " << format_str << " return  " << st.ToString();
+    if (!st.ok()) {
+        return Status::IOError(fmt::format("failed to list files in {}", format_str));
+    }
+    if (files.size() != 0) {
+        return Status::InternalError(fmt::format("dir {} is not empty", format_str));
+    }
+    auto res = object_store->delete_object(format_str);
+    LOG(WARNING) << "delete object " << format_str << " return  " << st.ToString();
+    return to_status(res);
+}
+
+Status StarletFileSystem::delete_dir_recursive(const std::string& dirname) {
+    auto uri = format_starlet_path(dirname);
+    auto format_str = format_starlet_path(uri);
```

Comment:
There are many unnecessary memory allocations and memory copies here, can we just remove the prefix "staros://" from `dirname` directly, without transforming a string to starlet::S3URI and back to string again?